### PR TITLE
[Merged by Bors] - feat port: Data.Int.Cast.Field

### PR DESCRIPTION
**Pull request by ChrisHughes24**
ee0c179cd3c8a45aa5bffbf1b41d8dbede452865

### CODE REVIEW

File: Mathlib/Data/Int/Cast/Field.lean [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+variable {α : Type _}
+
+/-- Auxiliary lemma for norm_cast to move the cast `-↑n` upwards to `↑-↑n`.
+
+(The restriction to `field` is necessary, otherwise this would also apply in the case where
+`R = ℤ` and cause nontermination.)
+-/
+@[norm_cast]
+theorem cast_neg_nat_cast {R} [Field R] (n : ℕ) : ((-n : ℤ) : R) = -n := by simp
```
Comment (fpvandoorn) [inline Mathlib/Data/Int/Cast/Field.lean:33]:
```suggestion
theorem cast_neg_natCast {R} [Field R] (n : ℕ) : ((-n : ℤ) : R) = -n := by simp
```
also in align below

File: Mathlib/Data/Int/Cast/Field.lean [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+namespace Int
+
+open Nat
+
+variable {α : Type _}
+
+/-- Auxiliary lemma for norm_cast to move the cast `-↑n` upwards to `↑-↑n`.
+
+(The restriction to `field` is necessary, otherwise this would also apply in the case where
```
Comment (fpvandoorn) [inline Mathlib/Data/Int/Cast/Field.lean:29]:
```suggestion
(The restriction to `Field` is necessary, otherwise this would also apply in the case where
```

**Review by fpvandoorn — COMMENTED** [review]
Thanks

bors d+

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**